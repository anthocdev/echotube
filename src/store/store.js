import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
