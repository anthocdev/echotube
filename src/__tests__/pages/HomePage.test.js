import React from "react";
import HomePage from "../../modules/Home";
import { Provider } from "react-redux";
import store from "../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Home Page Tests", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <HomePage />
          </Provider>
        </BrowserRouter>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <HomePage />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
  });
});
