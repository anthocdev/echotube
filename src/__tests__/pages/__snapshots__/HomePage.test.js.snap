// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Tests Renders Matches Snapshot 1`] = `
<div
  className="overlay"
>
  <div
    className="LogoContainer"
  >
    <svg
      className="Logo"
    >
      echotubelogo.svg
    </svg>
    <div
      className="subLogo"
    >
      Archive your songs, because YouTube will not.
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="homebox"
    >
      <h1>
         
        Welcome to EchoTube
        <p>
          You've reached an alternative environment to enjoy and archive your favorite songs from YouTube. Click the button below to get introduced!
        </p>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          href="/guide"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          style={
            Object {
              "backgroundColor": "rgba(153, 50, 204, 0.6)",
              "color": "white",
            }
          }
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Guide Page
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </h1>
    </div>
  </div>
</div>
`;
